cmake/CheckTypeRepresentation.cmake

```diff
@@ -0,0 +1,270 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
+# file Copyright.txt or https://cmake.org/licensing for details.
+
+#[=======================================================================[.rst:
+CheckTypeRepresentation
+-----------------------
+
+Check the bit representation of a type.
+
+.. command:: check_float_type_representation
+
+  .. code-block:: cmake
+
+    check_float_type_representation(<type> <var> [LANGUAGE <language>])
+
+  Check if the floating point type exists and determine its bit format.
+  ``<type>`` can be any floating point type supported by the compiler, but is
+  typically one of ``float``, ``double``, or ``long double``. If the type
+  exists, then the variable ``HAVE_<var>`` is set to true, otherwise it is set
+  to false. If the type exists, and ``check_float_type_representation()`` was
+  able to detect the bit representation of ``<type>``, then the variable
+  ``<var>`` is set to its name. If the type exists, but
+  ``check_float_type_representation()`` was not able to detect its type, then
+  the variable ``<var>`` is set to ``UNKNOWN``. If the type does not exist,
+  then ``<var>`` is set to an empty string.
+
+  If ``LANGUAGE`` is set, the specified compiler will be used to perform the
+  check. Acceptable values are ``C`` and ``CXX``.
+
+  ``check_float_type_representation()`` understands the following floating
+  point formats:
+
+  ``FLOAT_HP_IEEE754_LE``
+  ``FLOAT_HP_IEEE754_BE``
```

Comment:
I think the new naming scheme should still elude to the fact that it is a binary representation and not a decimal representation. How about:

`FLOAT_HP_B_IEEE754_LE`

Then potential decimal formats in the future could be:

`FLOAT_HP_D_IEEE754_LE`

Reply:
Rather than a subset of `FLOAT`, if it's a decimal representation then I'd rather just use a `DECIMAL` prefix instead of `FLOAT` since it really represents a different type with different math and different instructions.  I don't see a broad use case for this with ADIOS though so it sort of falls under the "TODO for feature completeness but probably won't" category.